netebpfext/net_ebpf_ext_sock_ops.c

```diff
@@ -552,3 +571,77 @@ net_ebpf_extension_sock_ops_flow_delete(uint16_t layer_id, uint32_t callout_id,
     if (attached_client != NULL)
         net_ebpf_extension_hook_client_leave_rundown(attached_client);
 }
+
+static uint64_t
+_ebpf_sock_ops_context_create(
+    _In_reads_bytes_opt_(data_size_in) const uint8_t* data_in,
+    size_t data_size_in,
+    _In_reads_bytes_opt_(context_size_in) const uint8_t* context_in,
+    size_t context_size_in,
+    _Outptr_result_maybenull_ void** context)
+{
+    NTSTATUS status;
+    bpf_sock_ops_t* sock_ops_context = NULL;
+
+    *context = NULL;
+
+    // This provider doesn't support data.
+    if (data_in != NULL || data_size_in != 0) {
+        status = STATUS_INVALID_PARAMETER;
+        goto Done;
+    }
+
+    // This provider requires context.
+    if (context_in == NULL || context_size_in < sizeof(bpf_sock_ops_t)) {
+        status = STATUS_INVALID_PARAMETER;
+        goto Done;
+    }
+
+    sock_ops_context =
+        (bpf_sock_ops_t*)ExAllocatePoolUninitialized(NonPagedPool, sizeof(bpf_sock_ops_t), NET_EBPF_EXTENSION_POOL_TAG);
+
+    if (sock_ops_context == NULL) {
+        status = STATUS_NO_MEMORY;
+        goto Done;
+    }
+
+    memcpy(sock_ops_context, context_in, sizeof(bpf_sock_ops_t));
+
+    *context = sock_ops_context;
+    sock_ops_context = NULL;
+    status = STATUS_SUCCESS;
+
+Done:
+    if (sock_ops_context != NULL) {
+        ExFreePool(sock_ops_context);
```

Comment:
can this use ebpf_free instead of directly using ExFreePool? (and same for the allocation)

Reply:
netebpfext.sys doesn't link against the ebpfcore platform layer.